Add Missing Python Version of TickDataFilteringAlgorithm (#7219)

hungnguyen9 opened the pull request:
#### Description
Missing Python Version of TickDataFilteringAlgorithm #7219 

#### Related Issue
Close #7219 

#### Requires Documentation Change
<!--- Please indicate if these changes will require updates to documentation, and if so, specify what changes are required -->
No

#### Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

AlexCatarino reviewed (changes requested):
Thank you for the contribution, @hungnguyen9 .

I ran this algorithm on QuantConnect Cloud and it raised the following exception:

```
During the algorithm initialization, the following exception has occurred: object does not implement ISecurityDataFilter
  at Initialize
```

hungnguyen9 commented:
> Thank you for the contribution, @hungnguyen9 .
> 
> I ran this algorithm on QuantConnect Cloud and it raised the following exception:
> 
> ```
> During the algorithm initialization, the following exception has occurred: object does not implement ISecurityDataFilter
>   at Initialize
> ```

hungnguyen9 commented:
> Thank you for the contribution, @hungnguyen9 .
> 
> I ran this algorithm on QuantConnect Cloud and it raised the following exception:
> 
> ```
> During the algorithm initialization, the following exception has occurred: object does not implement ISecurityDataFilter
>   at Initialize
> ```

Dear @AlexCatarino,

Thank you very much for your comment. I have modified the code. Could you please check again?

AlexCatarino commented on Algorithm.Python/TickDataFilteringAlgorithm.py:
```diff
@@ -0,0 +1,114 @@
+    # Save instance of the algorithm namespace
+    # </summary>
+    # <param name="algo"></param>
+    def ExchangeDataFilter(self, algo):
```
This should be `def __init__` and call `super()` this the class inherits from `SecurityDataFilter`. However, this class might not need to inherit from `SecurityDataFilter`.

AlexCatarino reviewed (dismissed):
Hi @hungnguyen9 ,

Please copy your algorithm, and run a backtest in QuantConnect Cloud. It should not raise exceptions. I ran the latest version and got the following:

```
During the algorithm initialization, the following exception has occurred: Trying to dynamically access a method that does not exist throws a TypeError exception. To prevent the exception, ensure each parameter type matches those required by the 'main.TickDataFilteringAlgorithm'>) method. Please checkout the API documentation.
  at Initialize
    self.Securities["SPY"].DataFilter = ExchangeDataFilter(self)
```

I recommend looking #7201 , since it has added the features that make this algorithm possible.

We have also added Documentation on:
https://www.quantconnect.com/docs/v2/writing-algorithms/securities/filtering-data

hungnguyen9 commented:
> Hi @hungnguyen9 ,
> 
> Please copy your algorithm, and run a backtest in QuantConnect Cloud. It should not raise exceptions. I ran the latest version and got the following:
> 
> ```
> During the algorithm initialization, the following exception has occurred: Trying to dynamically access a method that does not exist throws a TypeError exception. To prevent the exception, ensure each parameter type matches those required by the 'main.TickDataFilteringAlgorithm'>) method. Please checkout the API documentation.
>   at Initialize
>     self.Securities["SPY"].DataFilter = ExchangeDataFilter(self)
> ```
> 
> I recommend looking #7201 , since it has added the features that make this algorithm possible.
> 
> We have also added Documentation on: https://www.quantconnect.com/docs/v2/writing-algorithms/securities/filtering-data

Hi @AlexCatarino,

Thank you for your comment. I have modified my code, ran a backtest in QC cloud and it looked okay. Could you please check my PR again?

Regards,
Hung

Martin-Molinero reviewed (approved):
Thanks @hungnguyen9!
